rpm/0035-Add-TrimmedSubject-message-sort-key.patch

```diff
@@ -0,0 +1,108 @@
+     return QMailMessageSortKey(Subject, order);
+ }
+ 
++/*!
++    Returns a key that sorts messages by their subject without Re:, Fw: and Fwd: prefixes, according to \a order.
++
++    \sa QMailMessage::trimmedSubject()
++*/
++QMailMessageSortKey QMailMessageSortKey::trimmedSubject(Qt::SortOrder order)
++{
++    return QMailMessageSortKey(TrimmedSubject, order);
++}
++
+ /*!
+     Returns a key that sorts messages by their origination timestamp, according to \a order.
+ 
+diff --git a/src/libraries/qmfclient/qmailmessagesortkey.h b/src/libraries/qmfclient/qmailmessagesortkey.h
+index cfe9b82d..5072d9f9 100644
+--- a/src/libraries/qmfclient/qmailmessagesortkey.h
++++ b/src/libraries/qmfclient/qmailmessagesortkey.h
+@@ -66,7 +66,8 @@ public:
+         ListId,
+         RestoreFolderId,
+         RfcId,
+-        ParentThreadId
++        ParentThreadId,
++        TrimmedSubject
+     };
+ 
+     typedef QMailSortKeyArgument<Property> ArgumentType;
+@@ -97,6 +98,7 @@ public:
+     static QMailMessageSortKey sender(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey recipients(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey subject(Qt::SortOrder order = Qt::AscendingOrder);
++    static QMailMessageSortKey trimmedSubject(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey timeStamp(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey receptionTimeStamp(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey serverUid(Qt::SortOrder order = Qt::AscendingOrder);
+diff --git a/src/libraries/qmfclient/qmailstore_p.cpp b/src/libraries/qmfclient/qmailstore_p.cpp
+index 97f5020d..d14247df 100644
+--- a/src/libraries/qmfclient/qmailstore_p.cpp
++++ b/src/libraries/qmfclient/qmailstore_p.cpp
+@@ -632,6 +632,7 @@ static QMap<QMailMessageSortKey::Property, QMailMessageKey::Property> messageSor
+     map.insert(QMailMessageSortKey::Sender, QMailMessageKey::Sender);
+     map.insert(QMailMessageSortKey::Recipients, QMailMessageKey::Recipients);
+     map.insert(QMailMessageSortKey::Subject, QMailMessageKey::Subject);
++    map.insert(QMailMessageSortKey::TrimmedSubject, QMailMessageKey::TimeStamp);
+     map.insert(QMailMessageSortKey::TimeStamp, QMailMessageKey::TimeStamp);
+     map.insert(QMailMessageSortKey::ReceptionTimeStamp, QMailMessageKey::ReceptionTimeStamp);
+     map.insert(QMailMessageSortKey::Status, QMailMessageKey::Status);
+@@ -8572,7 +8573,16 @@ QMailStorePrivate::AttemptResult QMailStorePrivate::attemptQueryMessages(const Q
+                                                                          QMailMessageIdList *ids, 
+                                                                          ReadLock &)
+ {
+-    QSqlQuery query(simpleQuery(QLatin1String("SELECT id FROM mailmessages"),
++    bool sortByTrimmedSubject = false;
++    for (const QMailMessageSortKey::ArgumentType &arg : sortKey.arguments())
```

Comment:
Multiline for loop needs braces by qt coding style.

Reply:
Fixed.